tests/test_compression_lr_multiplier.py

```diff
@@ -0,0 +1,147 @@
+ limitations under the License.
+"""
+
+import contextlib
+import copy
+import pytest
+import torch
+import torch.nn as nn
+from torch.optim import SGD
+
+from nncf.layer_utils import CompressionParameter
+from tests.helpers import TwoConvTestModel, create_compressed_model_and_algo_for_test,\
+    get_grads, check_equal, check_not_equal
+from tests.quantization.test_algo_quantization import get_quantization_config_without_range_init
+from tests.sparsity.rb.test_algo import get_basic_sparsity_config
+
+
+@contextlib.contextmanager
+def torch_seed():
+    seed = torch.seed()
+    torch.manual_seed(0)
+    yield
+    torch.manual_seed(seed)
+
+
+def get_quantization_config():
+    config = get_quantization_config_without_range_init()
+    return config
+
+
+def get_sparsity_config():
+    config = get_basic_sparsity_config()
+    return config
+
+
+def modify_config(config, multiplier, multiplier_location, use_algo_list=False):
+    config = copy.deepcopy(config)
+
+    if use_algo_list:
+        config['compression'] = [config['compression']]
+
+    if multiplier_location in ['local', 'both']:
+        if isinstance(config['compression'], list):
+            for algo in config['compression']:
+                algo.update({
+                    'compression_lr_multiplier': multiplier
+                })
+        else:
+            config['compression'].update({
+                'compression_lr_multiplier': multiplier
+            })
+
+    if multiplier_location in ['global', 'both']:
+        config.update({
+            'compression_lr_multiplier': multiplier
+        })
+
+    return config
+
+
+def divide_params(params):
+    regular_params, compression_params = [], []
+    for param in params:
+        if param.requires_grad:
+            if isinstance(param, CompressionParameter):
+                compression_params.append(param)
+            else:
+                regular_params.append(param)
+
+    return regular_params, compression_params
+
+
+def make_train_step(config):
+    with torch_seed():
+        model = TwoConvTestModel()
+        for param in model.parameters():
+            if len(param.size()) > 1:
+                nn.init.xavier_normal_(param)
+            else:
+                nn.init.normal_(param)
+
+        model, compression_ctrl = create_compressed_model_and_algo_for_test(model, config)
+        optimizer = SGD(model.parameters(), lr=0.1)
+
+        x = torch.ones(config["input_info"]["sample_size"])
+        y = model(x)
+        loss = (y ** 2).sum()
+
+        loss.backward()
+        optimizer.step()
+
+    return model.parameters()
+
+
+@pytest.mark.parametrize("get_config", [
+    get_quantization_config,
+    get_sparsity_config
+])
+@pytest.mark.parametrize("multiplier", [0, 1, 10])
+@pytest.mark.parametrize("multiplier_location", ["local", "global", "both"])
+@pytest.mark.parametrize("use_algo_list", [True, False])
+def test_gradients(get_config, multiplier, multiplier_location, use_algo_list):
```

Comment:
Also possible to set up type hints here.

Reply:
`test_compression_lr_multiplier_affects_compression_param_gradients`

Reply:
Done.